### ISSUE TITLE
NavigationViewItems don't update the content frame when selected while using Narrator

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/terminal#12319]
### Windows Terminal version

1.12.10211.0

### Windows build number

22548.1002

### Other Software

Narrator

### Steps to reproduce

1. Open Narrator
2. Open settings UI
3. Navigate to an unselected navigation view item (i.e. "Actions")
4. Press <kbd>Space</kbd> or <kbd>Enter</kbd> to invoke it

### Expected Behavior

We should navigate to this item, and the content frame should update appropriately.

### Actual Behavior

The blue pill selects the item, but the content frame does not update.

In 1.12, the page title updates. In 1.13, the page title does not.

### Additional Details

PowerToys v0.55 settings has a NavigationView. It doesn't have this issue.
XAML Controls Gallery also doesn't have this issue.

Surprisingly, Narrator must be on for this bug to occur. If you do not have Narrator enabled, <kbd>Space</kbd> and <kbd>Enter</kbd> work just fine. I have not tested this with other screen readers.

**Comment by carlos-zamora** [comment]
This is _probably_ an issue we can file on https://github.com/microsoft/microsoft-ui-xaml, but we need to confirm that by...
- creating a sample app that experiences this issue
- filing an issue on their repo
- linking to that issue

Then, we can mark this as "Tracking-External".

**Comment by zadjii-msft** [comment]
> PowerToys v0.55 settings has a NavigationView. It doesn't have this issue.
> XAML Controls Gallery also doesn't have this issue.


Hmm. That might actually be on us then. PT Settings is also XAML Islands IIRC, so if their NavView works, then it's probably how we're using it.

**Comment by carlos-zamora** [comment]
You know what, this is a dup of #12003. Closing in favor of that one.
